static-site/main.tf

```diff
@@ -1,92 +1,91 @@
 // S3
-// static site bucket
-resource "aws_s3_bucket" "static_site" {
-  // name bucket after sub-domain name
-  bucket = "${var.sub_domain_name}"
-  policy = <<POLICY
-{
-  "Version":"2012-10-17",
-  "Statement":[
-    {
-      "Sid":"AddPerm",
-      "Effect":"Allow",
-      "Principal": "*",
-      "Action":["s3:GetObject"],
-      "Resource":["arn:aws:s3:::${var.sub_domain_name}/*"]
-    }
-  ]
-}
-POLICY
-
+// Site bucket
+resource "aws_s3_bucket" "site" {
+  // name bucket after domain name
+  bucket = "${var.domain_name}"
   website {
-    // bucket root index file
-    // subfolders are handled by a lambda@edge
     index_document = "index.html"
     error_document = "404.html"
```

Comment:
Error document should be configurable via the same variable as the cloudfront error document (add a leading slash to the cloudfront `response_page_path` if it needs it).

Reply:
After our convo, leaving this as is for now (after changing the error doc path to a variable).
AWS has some interesting insight about static S3 site + CF:
https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/custom-error-pages-response-code.html